rails-snow-queen/app/assets/javascripts/map.js

```diff
@@ -68,13 +68,15 @@ class Map {
   }
 
   addListeners() {
+    document.addEventListener("DOMContentLoaded", $('#addressSubmitModal').modal("show"));
     document.getElementById('AddressSearch').addEventListener('submit', this.handleSearchSubmit);
     this.drawingManager.addListener('polygoncomplete', this.handlePolygonCreated);
     this.removeControl.addEventListener('click', this.handleRemovePolygon);
   }
   
   handleSearchSubmit(event) {
     event.preventDefault();
+    $('#addressSubmitModal').modal("hide")
```

Comment:
Also probably doesn't belong in here. As much as you might hate it,  I think you should have an `AddressModal` class that takes care of these things and which has an `onSubmit` callback that you would wire up in `application.js` to call into a `setAddress` method on `Map` or something of that sort.